modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/NativeLibrariesFileDeletion.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.deployment.internal;
+
+import static java.lang.String.format;
+import static org.apache.commons.io.FileUtils.listFiles;
+import static org.mule.runtime.container.api.MuleFoldersUtil.getAppNativeLibrariesTempFolder;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.io.File;
+import java.util.Collection;
+
+import org.slf4j.Logger;
+
+public class NativeLibrariesFileDeletion {
+
+  private final String applicationName;
+  private final String[] nativeLibrariesExtensions = {"dylib", "dll", "jnilib", "so"};
+  private static final Logger LOGGER = getLogger(NativeLibrariesFileDeletion.class);
+
+  public NativeLibrariesFileDeletion(String applicationName) {
+    this.applicationName = applicationName;
+  }
+
+  public void doAction() {
+    File appNativeLibrariesTempFolder = getAppNativeLibrariesTempFolder(applicationName);
```

Comment:
We cannot delete all libs like this. Either we have a folder for this deployment in particular or we know the file names of the libraries that we must delete. Consider that the same app can be re-deployed in the meantime. Try looking at DeployableArtifactDescriptor and the MuleApplicationClassLoaderFactory (maybe we can add a native libs folder name at the descriptor and use it at the factory to create the NativeLibraryFinderFactory).